### PR TITLE
Feature/Prompt Fetch from Langsmith Hub

### PR DESCRIPTION
**Pull request by vinodkiran**

### CODE REVIEW

File: packages/server/src/index.ts [hunk 1]
```diff
@@ -1036,6 +1039,55 @@ export class App {
...
+                if (!credential) return res.status(404).json({ error: `Credential ${req.body.credential} not found` })
+                // Decrypt credentialData
+                const decryptedCredentialData = await decryptCredentialData(credential.encryptedData, credential.credentialName, undefined)
+
+                const headers = {}
+                // @ts-ignore
+                headers['x-api-key'] = decryptedCredentialData.langsmithApiKey
+
+                const tags = req.body.tags ? `tags=${req.body.tags}` : ''
+                const url = `https://web.hub.langchain.com/repos/?${tags}offset=0&limit=20&has_commits=true&sort_field=num_likes&sort_direction=desc&is_archived=false`
```
Comment (HenryHengZJ) [inline packages/server/src/index.ts:1080]: do we need offset and limit here? on the UI we dont have pagination right?

File: packages/ui/src/ui-component/dialog/PromptLangsmithHubDialog.js [hunk 2]
```diff
@@ -0,0 +1,475 @@
...
+            const createResp = await promptApi.getPrompt({
+                credential: credentialId,
+                promptName: selectedPrompt.full_name
+            })
+            if (createResp.data) {
+                prompt.detailed = createResp.data.templates
+            }
+        }
+        setSelectedPrompt(prompt)
+        await new Promise((resolve) => setTimeout(resolve, 500))
```
Comment (HenryHengZJ) [inline packages/ui/src/ui-component/dialog/PromptLangsmithHubDialog.js:163]: why do we need a setTimeout here

**Comment by HenryHengZJ** [comment]
Updated UI: 
image

image

File: packages/ui/src/ui-component/dialog/PromptLangsmithHubDialog.js (hunk 2 above)
Comment (vinodkiran) [reply packages/ui/src/ui-component/dialog/PromptLangsmithHubDialog.js]: Not needed, fixed and pushed changes

File: packages/server/src/index.ts (hunk 1 above)
Comment (vinodkiran) [reply packages/server/src/index.ts]: Not needed for now, fixed and pushed changes

**Pull request merged**